app/controllers/users/profil_controller.rb

```diff
@@ -12,8 +12,9 @@ def renew_api_token
     def update_email
       if @current_user.update(update_email_params)
         flash.notice = t('devise.registrations.update_needs_confirmation')
-      # to avoid leaking who has signed in
       elsif @current_user.errors&.details&.dig(:email)&.any? { |e| e[:error] == :taken }
+        UserMailer.account_already_taken(@current_user, requested_email).deliver_later
```

Comment:
rajouter un `current_user.update(unconfirmed_email: requestd_email)` pour éviter de leaker ?

Reply:
Je viens d'essayer : on prend une erreur de validation

```ruby
@current_user.update!(unconfirmed_email: requested_email)

ActiveRecord::RecordInvalid:
       La validation a échoué : Email déjà utilisé
```

J'imagine qu'il y a une contrainte d'unicité sur `email`+`unconfirmed_email`.

Comme ça, rapidement, je ne vois pas de solution évidente. Je suis chaud pou qu'on continue à se demander comment faire pour résoudre ce problème, et l'approche qu'on veut avoir – mais ça te va si on merge cette PR (qui est iso de ce point de vu là) et qu'on continue à itérer sur l'énumération des comptes ensuite ?

Reply:
yep